<launch>

  <arg name="odom_frame_id"   default="base_footprint"/>
  <arg name="base_frame_id"   default="base_footprint"/>
  <arg name="global_frame_id" default="map"/>
  <arg name="odom_topic" default="ground_truth/state" />
  <arg name="depth_topic" default="camera/depth/points" />
  <arg name="laser_topic" default="scan" />

  <node pkg="pointcloud_to_laserscan" type="pointcloud_to_laserscan_node" name="pointcloud_to_laserscan">
    <param name="min_height" value="0.3"/>
    <param name="max_height" value="1.7"/>
    <param name="range_min" value="0.2"/>
    <param name="target_frame" value="camera_link"/>

    <remap from="cloud_in" to="$(arg depth_topic)"/>
    <remap from="scan" to="$(arg laser_topic)"/>
  </node>

  <node pkg="move_base" type="move_base" respawn="false" name="move_base" output="screen">
    <rosparam file="$(find drone_slam)/param/costmap_common_params.yaml" command="load" ns="global_costmap" />
    <rosparam file="$(find drone_slam)/param/costmap_common_params.yaml" command="load" ns="local_costmap" />
    <rosparam file="$(find drone_slam)/param/local_costmap_params.yaml" command="load" />
    <rosparam file="$(find drone_slam)/param/global_costmap_params.yaml" command="load" />
    <rosparam file="$(find drone_slam)/param/teb_local_planner_params.yaml" command="load" />
    <rosparam file="$(find drone_slam)/param/move_base_params.yaml" command="load" />
    <rosparam file="$(find drone_slam)/param/global_planner_params.yaml" command="load" />
    <rosparam file="$(find drone_slam)/param/navfn_global_planner_params.yaml" command="load" />

    <!-- reset frame_id parameters using user input data -->
    <param name="global_costmap/global_frame" value="$(arg global_frame_id)"/>
    <param name="global_costmap/robot_base_frame" value="$(arg base_frame_id)"/>
    <param name="local_costmap/global_frame" value="$(arg odom_frame_id)"/>
    <param name="local_costmap/robot_base_frame" value="$(arg base_frame_id)"/>
    <param name="DWAPlannerROS/global_frame_id" value="$(arg odom_frame_id)"/>

    <remap from="cmd_vel" to="planned_cmd_vel"/>
    <remap from="odom" to="$(arg odom_topic)"/>
    <remap from="scan" to="$(arg laser_topic)"/>

  </node>
</launch>